cmd/queue/main_test.go

```diff
@@ -443,6 +445,56 @@ func TestQueueTraceSpans(t *testing.T) {
 	}
 }
 
+func BenchmarkProxyHandler(b *testing.B) {
+	params := queue.BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10}
+	breaker := queue.NewBreaker(params)
+	// Make the channel as big as possible to account for the largest b.N
+	reqChan := make(chan queue.ReqEvent, 10000000)
```

Comment:
Can we set this up like the actual queue-proxy would be setup? I.e. use `requestCountingQueueLength` and create a `queue.NewStats(time.Now(), reqChan, reportTicker.C, promStatReporter.Report)` with the actual prometheus reporter?

That way we correctly check the overhead of the request counting as well in both cases.

We should probably consider moving the handlers apart a little as proposed per #6803, but having a full-blown test of it all is useful as well.